src/test/test_bip32.py

```diff
@@ -533,33 +539,54 @@ def get_paths(path):
             ret = bip32_key_from_parent_path_str_n(m, path, len(path), wildcard, flags, key_out)
             self.assertEqual(ret, WALLY_EINVAL)
 
-        # After stripping the parents' private key, hardened path derivation fails
+        # Hardened derivation is possible from a full key
+        fn = lambda f: bip32_key_from_parent_path_str_n(m, 'm/1h/1h', 7, 0, f, key_out)
+        self.assertEqual(fn(FLAG_KEY_PRIVATE), WALLY_OK)
+        self.assertEqual(fn(FLAG_KEY_PUBLIC), WALLY_OK)
+        # After stripping the parents' private key, hardened derivation fails
         self.assertEqual(bip32_key_strip_private_key(m), WALLY_OK)
-        ret = bip32_key_from_parent_path_str_n(m, 'm/1h', len('m/1h'), 0, FLAG_KEY_PUBLIC, key_out)
-        self.assertEqual(ret, WALLY_EINVAL)
+        self.assertEqual(fn(FLAG_KEY_PRIVATE), WALLY_EINVAL)
+        self.assertEqual(fn(FLAG_KEY_PUBLIC), WALLY_EINVAL)
 
     def test_wildcard(self):
         master, pub, priv = self.create_master_pub_priv()
         m = byref(master)
-        flags = FLAG_STR_WILDCARD | FLAG_KEY_PRIVATE
         key_out, int_key_out = ext_key(), ext_key()
-        cases = [('m/1/*',    55),
-                 ('m/*',      55),
-                 ('m/1/*/1',  55),
-                 ('m/1/*h',   55),
-                 ('m/*h',     55),
-                 ('m/1/*h/1', 55)]
+        cases = [('m/1/*',     55),
+                 ('m/*',       55),
+                 ('m/1/*/1',   55),
+                 ('m/1h/*',    55),
+                 ('m/1h/*/1',  55),
+                 ('m/1h/*/1h', 55),
+                 ('m/1/*h',    55),
+                 ('m/*h',      55),
+                 ('m/1/*h/1',  55),
+                 ('m/1/*h/1h', 55)]
```

Comment:
Could we add here something like: `m/*h/1h/1h/1/1`, as that covers the bip44/48/84 schemes - common singlesig use cases.  Cheers.